src/workloads/execution/FilterWithComplexLogicalExpression.yml

```diff
@@ -32,7 +32,108 @@ GlobalDefaults:
   DocumentCount: &DocumentCount 1e6
   Repeat: &Repeat 50
   Threads: &Threads 1
-  MaxPhases: &MaxPhases 17
+  MaxPhases: &MaxPhases 25
+
+# Set of actor templates for executing a rooted $and/$or over a configurable number of clauses for
+# the aggregation and match languages, respectively.
+ActorTemplates:
+- TemplateName: AggregationOrWithManyChildClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "AggregationOr"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter:
+                {$expr:
+                   {$or:
+                      {^Array: {of: {$and: [
+                        {$eq:
+                           ["$a",
+                            {^RandomInt:
+                               {min: 2, max: { ^Parameter: { Name: "aMaxValue", Default: 1000}}}}]},
+                        {$eq:
+                           ["$b",
+                            {^RandomInt:
+                               {min: 2, max: {^Parameter: { Name: "bMaxValue", Default: 1000}}}}]}]},
+                                number: {^Parameter: { Name: "NumberOfClauses", Default: 50}}}}}}
+
+- TemplateName: MatchOrWithManyChildClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "MatchOr"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter:
+                { $or:
+                    { ^Array:
+                        { of:
+                            { a: { ^RandomInt:
+                                     { min: 2, max: { ^Parameter: { Name: "aMaxValue", Default: 1000 } } } },
+                              b: { ^RandomInt:
+                                     { min: 2, max: { ^Parameter: { Name: "bMaxValue", Default: 1000 } } } } },
+                          number: { ^Parameter: { Name: "NumberOfClauses", Default: 50 } } } } }
+
+- TemplateName: MatchAndWithManyChildClausesTemplate
```

Comment:
Note that I did this only for the match language. I could do this for the agg expressions, but I'm not sure what this would look like (outside of doing a wide document style query, that is, {$and: [{$eq: ["$a", 1]}, {"$b", 1}, ...]})